src/System.Private.ServiceModel/tests/Common/Scenarios/ScenarioTestTypes.cs

```diff
@@ -803,4 +831,26 @@ public void OnPingCallback(Guid guid)
             _tcs.SetResult(guid);
         });
     }
+
+    public void OnXmlPingCallback(XmlCompositeType xmlCompositeType)
+    {
+        // Set the result in an async task with a 100ms delay to prevent a race condition
+        // where the OnPingCallback hasn't sent the reply to the server before the channel is closed.
+        Task.Run(async () =>
+        {
+            await Task.Delay(100);
+            _xml_tcs.SetResult(xmlCompositeType);
```

Comment:
Just a question -- why 100ms?  Don't block on this, I was just wondering if there is some other way to detect when it's okay to set the TCS.